### ISSUE TITLE
Breaking Change with spread params in 5.5

### ISSUE THREAD

**Issue by maphe** [mikro-orm/mikro-orm#4113]
**Describe the bug**
Prior to v5.5, it was possible to use spread params in functions like `Collection.add`.

```ts
const items = [a, b, c];
collection.add(...items);
```
This is broken after upgrading to 5.5 with the following error: `TS2556: A spread argument must either have a tuple type or be passed to a rest parameter.`

**Expected behavior**
Either the doc should explicitly point out the breaking change, or the code should not break.

**Additional context**
I know this is fixable by changing to
```ts
const items = [a, b, c];
collection.add(items);
```
and that might be on purpose that the old way is not supported anymore, I just think it should be obvious from the doc/changelog that upgrading may break existing code.

Also I did not look for an exhaustive list of functions that broke, `Collection.add` and `Collection.remove` are just the 2 I ran into.

**Versions**

| Dependency | Version |
| - | - |
| node | `v18.14.2` |
| typescript | `4.5.9` |
| mikro-orm | `5.5.3` |
| your-driver | `MySQL` |

**Comment by B4nan** [comment]
This comes from a performance fix (#3573), doing `...items` was very inefficient, and you should replace that code anyway, using the first argument as an array is _much_ more efficient.

Note that it is a TS error, not a runtime error, technically it is still possible to do the same. There were more slightly breaking changes on type level like this, while I try to save them for major releases, sometimes it needs to be done sooner.

Will add a note about this to the 5.5 release notes, as you can see from the PR, it was already mentioned there, so not an unwanted side effect.

**Comment by maphe** [comment]
Perfect, thank you.
